src/http/client.c

```diff
@@ -766,7 +766,8 @@ int http_client_set_tls_hostname(struct http_cli *cli,
 	if (!cli || !hostname)
 		return EINVAL;
 
-	return tls_set_hostname(cli->tls_hostname, hostname);
+	cli->tls_hostname = mem_deref(cli->tls_hostname);
```

Comment:
Should this `mem_ref`?

Reply:
No, because the following pl_strdup() will allocate a new buffer for cli->tls_hostname. The previous has to be freed.

Here is a bug in first parameter pointer notation. But in the meanwhile I was thinking again about the necessity of this function. It should encapsulate the OPENSSL_VERSION_NUMBER check in openssl/tls.c. But this is done later again during initialization of the TLS handshake in tls_peer_set_verify_host(). --> I'll set this PR to draft, maybe delete later.

Currently I am working on a http request PR for libre that supports basic/digest authorization and TLS on top of http_cli. Then maybe I'll drop tls_set_hostname().

r,c.

Reply:
Ah I see. Thanks for clarification.